PlayKitSmartSwitch/Classes/SmartSwitchMediaEntryInterceptor.swift

```diff
@@ -72,10 +74,23 @@ extension SmartSwitchMediaEntryInterceptor: PKMediaEntryInterceptor {
         }
     }
     
-    private func getOrderedCDN(originalURL: URL, completion: @escaping (_ cdn: SmartSwitcCDNItem?, _ error: Error?) -> Void) {
-        guard let request: KalturaRequestBuilder = KalturaRequestBuilder(url: self.config.smartSwitchUrl,
+    private func getOrderedCDN(originalURL: URL,
+                               completion: @escaping (_ cdn: Provider?, _ error: Error?) -> Void) {
+        
+        serverURL = serverURL.replacingOccurrences(of: "{domainUrl}", with: self.config.domainUrl)
+        serverURL = serverURL.replacingOccurrences(of: "{accountCode}", with: self.config.accountCode)
+        
+        if let application = self.config.application, !application.isEmpty {
+            serverURL = serverURL.replacingOccurrences(of: "{application}",
+                                                       with: application.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowedCharacterSet) ?? application)
+        } else {
+            serverURL = serverURL.replacingOccurrences(of: "{application}", with: "default")
+        }
+        
+        guard let request: KalturaRequestBuilder = KalturaRequestBuilder(url: serverURL,
                                                                          service: nil,
                                                                          action: nil) else {
+            PKLog.error("Can not create get ordered CDN request.")
             completion(nil, nil)
```

Comment:
We need to return an error here.

Reply:
Not sure how users will handle this error and if this information will help them some how.
I will add it